api/v1alpha1/clusterextension_types.go

```diff
@@ -130,6 +133,7 @@ type ClusterExtensionInstallConfig struct {
 	//+kubebuilder:validation:Pattern:=^[a-z0-9]([-a-z0-9]*[a-z0-9])?$
 	//+kubebuilder:validation:MaxLength:=63
 	//+kubebuilder:validation:XValidation:rule="self == oldSelf",message="namespace is immutable"
+	//+kubebuilder:validation:Required
 	Namespace string `json:"namespace"`
```

Comment:
There were a few comments from Joel related to the GoDoc for this field that I don't see reflected here. Did we decide against accepting those suggestions or making adjustments to clarify some of the questions asked?

Ref:
- https://github.com/openshift/api/pull/2067/files#r1812416875
- https://github.com/openshift/api/pull/2067/files#r1812425452
- https://github.com/openshift/api/pull/2067/files#r1812430084
- https://github.com/openshift/api/pull/2067/files#r1812416228

Reply:
Looks like most of the comments are addressed . Also https://github.com/openshift/api/pull/2067/files#r1812416228 is an optional change. Either way it is fine.

Reply:
Yeah, @everettraven was way ahead of my WIP PR here. 😄 
I think I've caught up now, and took the 'WIP' off for a full review.